src/databricks/sql/client.py

```diff
@@ -433,6 +439,43 @@ def _determine_parameter_approach(
         else:
             return ParameterApproach.NATIVE
 
+    def _only_dbsql_parameters_in_list(self, params: ListOfParameters) -> bool:
+        """Return True if all members of the list are DbsqlParameter instances"""
+        return all([isinstance(i, DbsqlParameter) for i in params])
+
+    def _all_dbsql_parameters_are_named(self, params: List[DbsqlParameter]) -> bool:
+        """Return True if all members of the list have a non-null .name attribute"""
+        return all([i.name is not None for i in params])
+
+    def _list_of_params_can_use_named_structure(self, params: ListOfParameters) -> bool:
+        """A list of parameters can use the named ParameterStructure if every member of
+        the list is a DbsqlParameter type and every member has a non-null .name attribute.
+
+        Otherwise, the list can only use the positional ParameterStructure
+        """
+
+        return self._only_dbsql_parameters_in_list(
+            params
+        ) and self._all_dbsql_parameters_are_named(params)
+
+    def _determine_parameter_structure(
+        self,
+        parameters: Optional[Union[ListOfParameters, DictOfParameters]],
+    ) -> ParameterStructure:
+
+        if parameters is None:
+            return ParameterStructure.NONE
+
+        if not isinstance(parameters, (list, dict, tuple)):
```

Comment:
this method is not part of the public interface, so in theory, your code is the only thing invoking it.  In such cases, I would say its being very paranoid/defensive to validate your types at runtime.  Better to spend your effort ensuring that you use type assignments extensively, and rely on mypy to ensure you only call this method with a valid type.